### DISCUSSION TITLE
Devcontainer

### DISCUSSION THREAD

**Discussion by jasonpnnl** [open-webui/open-webui#5531] [Ideas]
For our own purposes I add a devcontainer in our forked repo. We use devcontainers instead of virtual environments or conda. I would be willing to contribute the devcontainer to the upstream repo if that is of interest.

**Comment by mfgering** [comment]
I *am* interested in your devcontainer! I currently debug the source code using a FastAPI python config, but I'd prefer learning how to do this with devcontainers.

**Comment by selenecodes** [comment]
@jasonpnnl do you still have that devcontainer?

**Reply by gmacario** [reply]
Hello,

I made a PR https://github.com/open-webui/open-webui/pull/15543 with the configuration I am currently using.
Unfortunately it was not accepted

**Reply by guenhter** [reply]
DevContainers are awesome and I use them all the time for Node and Python for OWUI but I completely understand why such PRs are not accepted. Here a few of why I think this is not accepted:
* It is specific to a single IDE
* The community might never can agree on what is the best set of predefined plugins or what is the best way of how the container is created
* If someone already has a .devcontainer folder with his personal settings, the `.devcontainer` in git would then override the personalized one

This are just a few thoughts.

What I do:

I have this content in the `.git/info/exclude ` file

```
.devcontainer
.vscode
```

And can therefore use personalised settings for those two folders.  

Nevertheless, it would be great to have this documented. Maybe you can open a PR in the docs (e.g. in or under https://docs.openwebui.com/getting-started/advanced-topics/development) and add your example devcontainer config there.
